lms/static/scripts/frontend_apps/components/TruncatedText.tsx

```diff
@@ -0,0 +1,49 @@
+import type { JSX } from 'preact';
+import { useEffect, useRef, useState } from 'preact/hooks';
+
+export type TruncatedTextProps = JSX.HTMLAttributes<HTMLSpanElement> & {
+  children: string;
+};
+
+/**
+ * A text container that elides its content.
+ *
+ * If the text overflows, the full content is added as a title for the
+ * container.
+ */
+export default function TruncatedText({
+  children,
+  title,
+  ...htmlAttrs
+}: TruncatedTextProps) {
+  const [overflow, setOverflow] = useState(false);
+  const ref = useRef<HTMLSpanElement | null>(null);
+
+  useEffect(() => {
+    const element = ref.current;
+    if (!element) {
+      return () => {};
+    }
+
+    const computeIsOverflowing = () =>
+      setOverflow(element.scrollWidth > element.clientWidth);
+
+    // Check if element is overflowing on initial render
+    computeIsOverflowing();
+    // Re-check when the element intersects with the viewport
```

Comment:
Since the listboxes are scrollable, I need to check if the `ResizeObserver` works when the overflowing items are our outside the visible part.

Reply:
What was the conclusion here? I wouldn't expect it to matter whether the text items are in the visible region of the scroll box or not, since their size should be the same.

Reply:
Ouch! I got distracted with something else and ended up forgetting to test this 😅. Give me a minute.

Reply:
The ResizeObserver works as well, and the callback runs less frequently. I have pushed the changes.